src/scenic/core/vectors.py

```diff
@@ -4,427 +4,659 @@
-		ro = offset.rotatedBy(heading)
-		return self + ro
-
-	@vectorOperator
-	def offsetRadially(self, radius, heading) -> Vector:
-		return self.offsetRotated(heading, Vector(0, radius))
-
-	@scalarOperator
-	def distanceTo(self, other) -> float:
-		if not isinstance(other, Vector):
-			return other.distanceTo(self)
-		dx, dy = other.toVector() - self
-		return math.hypot(dx, dy)
-
-	@scalarOperator
-	def angleTo(self, other) -> float:
-		dx, dy = other.toVector() - self
-		return normalizeAngle(math.atan2(dy, dx) - (math.pi / 2))
-
-	@scalarOperator
-	def angleWith(self, other) -> float:
-		"""Compute the signed angle between self and other.
-
-		The angle is positive if other is counterclockwise of self (considering
-		the smallest possible rotation to align them).
-		"""
-		x, y = self.x, self.y
-		ox, oy = other.x, other.y
-		return normalizeAngle(math.atan2(oy, ox) - math.atan2(y, x))
-
-	@scalarOperator
-	def norm(self) -> float:
-		return math.hypot(*self.coordinates)
-
-	@scalarOperator
-	def dot(self, other) -> float:
-		x, y = self.x, self.y
-		ox, oy = other.x, other.y
-		return (x * ox) + (y * oy)
-
-	@vectorOperator
-	def normalized(self) -> Vector:
-		l = math.hypot(*self.coordinates)
-		return Vector(*(coord/l for coord in self.coordinates))
-
-	@vectorOperator
-	def __add__(self, other) -> Vector:
-		return Vector(self[0] + other[0], self[1] + other[1])
-
-	@vectorOperator
-	def __radd__(self, other) -> Vector:
-		return Vector(self[0] + other[0], self[1] + other[1])
-
-	@vectorOperator
-	def __sub__(self, other) -> Vector:
-		return Vector(self[0] - other[0], self[1] - other[1])
-
-	@vectorOperator
-	def __rsub__(self, other) -> Vector:
-		return Vector(other[0] - self[0], other[1] - self[1])
-
-	@vectorOperator
-	def __mul__(self, other) -> Vector:
-		return Vector(*(coord*other for coord in self.coordinates))
-
-	def __rmul__(self, other) -> Vector:
-		return self.__mul__(other)
-
-	@vectorOperator
-	def __truediv__(self, other) -> Vector:
-		return Vector(*(coord/other for coord in self.coordinates))
-
-	def __len__(self):
-		return len(self.coordinates)
-
-	def __getitem__(self, index):
-		return self.coordinates[index]
-
-	def __repr__(self):
-		return f'({self.x} @ {self.y})'
-
-	def __eq__(self, other):
-		if isinstance(other, Vector):
-			return other.coordinates == self.coordinates
-		elif isinstance(other, (tuple, list)):
-			return tuple(other) == self.coordinates
-		else:
-			return NotImplemented
-
-	def __hash__(self):
-		return hash(self.coordinates)
-
-	@classmethod
-	def encodeTo(cls, vec, stream):
-		stream.write(struct.pack('<dd', *vec.coordinates))
-
-	@classmethod
-	def decodeFrom(cls, stream):
-		return cls(*struct.unpack('<dd', stream.read(16)))
+    """A 3D vector, whose coordinates can be distributions."""
+    def __init__(self, x, y, z=0):
+        self.coordinates = (x, y, z)
+        super().__init__(self.coordinates)
+
+    @classmethod
+    def fromSpherical(cls, vector):
```

Comment:
This method is never used, and I don't think we should encourage storing spherical coordinates in a `Vector` anyway, since nearly all of the vector operators assume the coordinates are Cartesian. I'd suggest removing this method, renaming `cartesianToSpherical` to just `sphericalCoordinates` (you could even make it a `@property`), and having that return an ordinary tuple.